sdk/cosmos/azure-cosmos/azure/cosmos/_execution_context/aio/execution_dispatcher.py

```diff
@@ -70,12 +81,8 @@ async def __anext__(self):
         try:
             return await self._execution_context.__anext__()
         except CosmosHttpResponseError as e:
-            if _is_partitioned_execution_info(e):
-                query_to_use = self._query if self._query is not None else "Select * from root r"
-                query_plan_dict = await self._client._GetQueryPlanThroughGateway(
-                    query_to_use, self._resource_link, self._options.get('excludedLocations'))
-                query_execution_info = _PartitionedQueryExecutionInfo(query_plan_dict)
-                self._execution_context = await self._create_pipelined_execution_context(query_execution_info)
+            if _is_partitioned_execution_info(e) or _is_hybrid_search_query(self._query, e):
```

Comment:
QQ: so for query accepted by gateway, it will not be populated with session token?